### PR TITLE
user sync utilities

### LINKED ISSUE #212: cookieSync code redundant; too noisy
**Describe the bug**

**To Reproduce**
https://github.com/prebid/prebid-universal-creative/blob/master/src/cookieSync.js is far too noisy and shares lots of code with https://github.com/prebid/prebid-universal-creative/blob/master/src/cookieSyncWithConsent.js . They should not have code copy pastes and they should not console log so much. 

load_cookie_with_consent.html and load_cookie.html should source the same js and be identical in any other way.

### LINKED ISSUE #226: Feature Request: eject load-cookie.html from the PUC repo, form new repo
eject the cookiesync and load-cookie logic from the puc; it is unrelated and we would like to be able to build and release one without the other.

### PR DESCRIPTION
**Pull request by dgirardi**
Reimplementation of `load-cookie` and `load-cookie-with-consent`, from a single codebase, and adding GPP support.

GPP support is limited to accepting and relaying the `gpp_sid` and `gpp` query string parameters; there's currently no way to poll AMP for GPP.

Closes https://github.com/prebid/prebid-universal-creative/issues/212
Closes https://github.com/prebid/prebid-universal-creative/issues/226

Related: https://github.com/prebid/user-sync/issues/2

### CODE REVIEW

**Review by spotxslagle — APPROVED** [review]
LGTM

File: src/params.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            params.log?.(err, e);
+            if (!required) {
+                return params;
+            } else {
+                throw new Error(err);
+            }
+        });
+}
+
+export function resolveParams(params, alwaysPollAMP = false, resolveConsent = resolveConsentParams) {
```
Comment (spotxslagle) [inline src/params.js:92]: I approved on the assumption that everything is working as intended, but I am a bit curious why the resolveConsent var is used as a third parameter. As far as I can tell, resolveParams is only used by load syncs, which doesn't allow third argument to be passed in. I assume this is a placeholder for future development or something similar, but it did seem worth commenting on.

Comment (dgirardi) [reply src/params.js:92]: You're right it's unnecessary. I think at some point this function was more complex and I needed to mock this out from unit tests.

**Comment by bretg** [comment]
Let me see if I can summarize what this all about:

1. You guys don't like load-cookie in the PUC repo
2. So it's being banished to this separate repo
3. Which means that companies supporting AMP need to be alerted to this move and upgrade their procedures to pull from 2 repos where they formerly dealt with one.

The README is pretty good. We'll just need to upgrade the AMP docs. People are likely to miss this shift because there's no such thing as a "major release" in this area.

**Review by bretg — APPROVED** [review]

**Pull request merged**